### PR TITLE
drivers: Intel: dmic: make DMIC prm and active fifo count shared

### PR DESCRIPTION
**Pull request by ranj063**
When the DMIC pipeline is moved to a core other than the
primary code, the DMIC params and dmic_active_fifos are
not valid. Modify them to be shared data and handle
cache management when accessing/modifying them.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

### CODE REVIEW

File: src/drivers/intel/dmic.c [hunk 1]
```diff
@@ -1624,7 +1627,7 @@ static int dmic_remove(struct dai *dai)
 	interrupt_unregister(dmic->irq, dai);
 
 	/* The next end tasks must be passed if another DAI FIFO still runs */
-	if (dmic_active_fifos)
+	if (uncached_dmic_active_fifos)
```
Comment (fuyuntsuo) [inline src/drivers/intel/dmic.c:1630]: Missing asterisk here `(*uncached_dmic_active_fifos)`

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@ranj063 good find, I'll cherry-pick for rc2

**Comment by keyonjie** [comment]
Hi @ranj063 we are still seeing DSP panic with this, looks we need to use uncached mode of dmic_prm[] also.
@aiChaoSONG to confirm.

**Comment by keyonjie** [comment]
@ranj063 @lgirdwood one more change needed per my observation:
https://github.com/thesofproject/sof/pull/4196

**Pull request merged**